xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/AbstractDocumentListClassPropertyValuesProvider.java

```diff
@@ -58,6 +61,17 @@
     @Named("viewable")
     private QueryFilter viewableFilter;
 
+    /**
+     * A map with an icon css class / url or with {@link org.xwiki.icon.IconManager#getMetaData icon metadata}.
+     *
+     * <p>The entries of this map are meant to be added to the metadata of the
+     * {@link #getValueFromQueryResult values of the query result}.
+     *
+     * @param documentReference document reference used to generate the icon
+     * @return a map with icon related information
+     */
+    protected abstract Map<String, Object> getIcon(DocumentReference documentReference);
```

Comment:
I'm not convinced we should support both types of icon representations. It complicates the code, especially ``xwiki.selectize.js``. I think I prefer to return only ``META_DATA_ICON`` having a map as value. The user and group providers can fill the icon meta data map with the avatar URL and the ``IMAGE`` icon type. WDYT?

Reply:
If I understand you want to move the old icon key (`META_DATA_ICON`) into the new icon metadata map (`META_DATA_ICON_META_DATA`) so that we only keep the map? It will make `xwiki.selectize.js` simpler indeed but we have to make sure that there are no other files to change.